### PR TITLE
[Bugfix](datetime) fix DateLiteral range check is no longer valid

### LINKED ISSUE #11914: [Bug] DateLiteral range check is no longer valid
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

DateLiteral range check is no longer valid  ，while in our docs 
```
DATETIME Date and time type, value range is ['0000-01-01 00:00:00','9999-12-31 23:59:59']. The form of printing is 'YYYY-MM-DD HH:MM:SS'

```

### What You Expected?

range check is in use

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yangzhg**
# Proposed changes

Issue Number: close #11914 

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
lgtm

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java [hunk 1]
```diff
@@ -435,6 +435,9 @@ private void init(String s, Type type) throws AnalysisException {
                 this.roundFloor(((ScalarType) type).getScalarScale());
             }
             this.type = type;
+            if (checkRange() || checkDate()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java:438]: Should it be:
```
if (!checkRange() || !checkDate())
```
?

Comment (yangzhg) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java:438]: `checkRange` and `checkDate` return true is error 😂
https://github.com/yangzhg/doris/blob/881670566c0aa577dd83af69c7b3d9f7a3986ab2/fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java#L1298

**Pull request merged**